src/Sylius/Bundle/ApiBundle/spec/ApiPlatform/ResourceMetadataPropertyValueResolverSpec.php

```diff
@@ -0,0 +1,205 @@
+            ]
+        )->shouldReturn(
+            [
+                'admin_get' => [
+                    'method' => 'GET',
+                    'path' => 'admin/path/{tokenValue}'
+                ],
+                'admin_post' => [
+                    'method' => 'POST',
+                    'path' => 'admin/path'
+                ]
+            ]
+        );
+    }
+
+    public function it_returns_child_config_if_parent_config_is_null(ApiResourceConfigurationMergerInterface $apiResourceConfigurationMerger): void
+    {
+        $resourceMetadata = new ResourceMetadata(
+            null,
+            null,
+            null,
+            null
+        );
+
+        $apiResourceConfigurationMerger->mergeConfigs(
+            Argument::any()
+        )->shouldNotBeCalled();
+
+        $this->resolve(
+            'itemOperations',
+            $resourceMetadata,
+            [
+                'itemOperations' => [
+                    'admin_patch' => [
+                        'method' => 'PATCH',
+                        'path' => 'admin/path'
+                    ]
+                ]
+            ]
+        )->shouldReturn(
+            [
+                'admin_patch' => [
+                    'method' => 'PATCH',
+                    'path' => 'admin/path'
+                ]
+            ]
+        );
+    }
+
+    public function it_overwrites_parent_config(ApiResourceConfigurationMergerInterface $apiResourceConfigurationMerger): void
+    {
+        $resourceMetadata = new ResourceMetadata(
+            'parent_short_name',
+            null,
+            null,
+        );
+
+        $apiResourceConfigurationMerger->mergeConfigs(
+            Argument::any()
+        )->shouldNotBeCalled();
+
+        $this->resolve(
+            'shortName',
+            $resourceMetadata,
+            [
+                'shortName' => 'child_short_name'
+            ]
+        )->shouldReturn('child_short_name');
+    }
+
+    public function it_throws_error_if_parent_data_is_array_and_child_type_is_different(ApiResourceConfigurationMerger $apiResourceConfigurationMerger): void
+    {
+        $resourceMetadata = new ResourceMetadata(
+            'parent_short_name',
+            null,
+            null,
+            null,
+            [
+                'admin_get' => [
+                    'method' => 'GET',
+                    'path' => 'admin/path/{tokenValue}'
+                ],
+                'admin_post' => [
+                    'method' => 'POST',
+                    'path' => 'admin/path'
+                ]
+            ]
+        );
+
+        $apiResourceConfigurationMerger->mergeConfigs(
+            Argument::any()
+        )->shouldNotBeCalled();
+
+        $this->shouldThrow(\InvalidArgumentException::class)->during('resolve', [
+            'collectionOperations',
+            $resourceMetadata,
+            [
+                'collectionOperations' => 'invalid_string'
+            ]
```

Comment:
Lines splitting are done quite aggressively in some places 😄 Could be just:
```php
['collectionOperations' => 'invalid_string']
```
here, or
```php
$apiResourceConfigurationMerger->mergeConfigs(Argument::any())->shouldNotBeCalled();
```
above 🚀